dashboard/app/views/levels/_reference_area.html.haml

```diff
@@ -1,7 +1,10 @@
 #reference_area.reference_area
   - if can_view_solution?
     .solution_link.hide-as-student
-      = link_to t('reference_area.teacher'), build_script_level_path(@script_level, solution: true)
+      - if params[:solution] != "true"
```

Comment:
nit: this should probably be just `if !params[:solution]`, since that's all we actually check for when deciding whether or not to show the solution